### PR TITLE
Support updating data source values

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/webstudio-is/webstudio-builder/issues/1529

Added data source values store which can be updated from both components and props panel.

```js
const { setState } = useContext(ReactSdkContext)
const instanceId = getInstanceIdFromComponentProps(props)

setState(instanceId, propName, value)
```

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - test nothing is broken in props panel on preview
- [ ] hi @istarkov, I need you to do
  - detailed review (read every line)

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: packages/react-sdk/src/context.tsx [hunk 1]
```diff
@@ -35,11 +36,21 @@ export const ReactSdkContext = createContext<
     propsByInstanceIdStore: ReadableAtom<PropsByInstanceId>;
     assetsStore: ReadableAtom<Assets>;
     pagesStore: ReadableAtom<Pages>;
+    dataSourceValuesStore: ReadableAtom<Map<DataSource["id"], unknown>>;
+    setState: (
```
Comment (kof) [inline packages/react-sdk/src/context.tsx:40]: should it be just setState? it may be better when more specific? e.g. setInstanceState?

Comment (TrySound) [reply packages/react-sdk/src/context.tsx:40]: setDataSourceValue?

**Review by kof — APPROVED** [review]

File: packages/react-sdk/src/context.tsx (hunk 1 above)
Comment (kof) [reply packages/react-sdk/src/context.tsx:40]: yeah, that's good

File: apps/builder/app/canvas/canvas.tsx [hunk 2]
```diff
@@ -102,6 +104,24 @@ const useElementsTree = (components: Components, params: Params) => {
...
+            newValues.set(dataSourceId, dataSource.value);
+          }
+          for (const [dataSourceId, dataSourceValue] of dataSourceValues) {
+            newValues.set(dataSourceId, dataSourceValue);
+          }
+          return newValues;
+        }
+      ),
+      onDataSourceUpdate: (dataSourceId, value) => {
+        const dataSourceValues = new Map(dataSourceValuesStore.get());
```
Comment (istarkov) [inline apps/builder/app/canvas/canvas.tsx:121]: Looks like its only to force recompute? To emulate immutable data structure? Mb use immer instead?

**Review by istarkov — APPROVED** [review]

File: apps/builder/app/canvas/canvas.tsx (hunk 2 above)
Comment (TrySound) [reply apps/builder/app/canvas/canvas.tsx:121]: No big deal to use immer here

**Pull request merged**